tests/mu_define_enum_test.c

```diff
@@ -34,5 +66,62 @@ int run_mu_define_enum_tests(void)
 
     POOR_MANS_ASSERT(MU_ENUM_VALUE_COUNT(TEST4_ENUM_VALUES) == 3);
 
+    const char* TEST5_test5_b = MU_ENUM_TO_STRING(TEST5_ENUM, 42);
+    POOR_MANS_ASSERT(TEST5_test5_b != NULL);
+    POOR_MANS_ASSERT(strcmp("test5_b = 42", TEST5_test5_b) == 0);
+
+    const char* TEST5_unknown = MU_ENUM_TO_STRING(TEST5_ENUM, 41);
+    POOR_MANS_ASSERT(TEST5_unknown != NULL);
+    POOR_MANS_ASSERT(strcmp("UNKNOWN", TEST5_unknown) == 0);
+
+    const char* TEST6_test6_a = MU_ENUM_TO_STRING(TEST6_ENUM, test6_a);
+    POOR_MANS_ASSERT(TEST6_test6_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_a = 42", TEST6_test6_a) == 0);
+
+    // test6_b and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_b = MU_ENUM_TO_STRING(TEST6_ENUM, test6_b);
+    POOR_MANS_ASSERT(TEST6_test6_b != NULL);
+    POOR_MANS_ASSERT(strcmp("TEST6_ENUM_INVALID", TEST6_test6_b) == 0);
+
+    const char* TEST6_test6_c = MU_ENUM_TO_STRING(TEST6_ENUM, test6_c);
+    POOR_MANS_ASSERT(TEST6_test6_c != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_c = -1", TEST6_test6_c) == 0);
+
+    // test6_d and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_d = MU_ENUM_TO_STRING(TEST6_ENUM, test6_d);
+    POOR_MANS_ASSERT(TEST6_test6_d != NULL);
+    POOR_MANS_ASSERT(strcmp("TEST6_ENUM_INVALID", TEST6_test6_d) == 0);
+
+    const char* TEST6_test6_e = MU_ENUM_TO_STRING(TEST6_ENUM, test6_e);
+    POOR_MANS_ASSERT(TEST6_test6_e != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_e", TEST6_test6_e) == 0);
+
+    // test6_e and test6_f are both the same value
+    const char* TEST6_test6_f = MU_ENUM_TO_STRING(TEST6_ENUM, test6_f);
+    POOR_MANS_ASSERT(TEST6_test6_f != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_e", TEST6_test6_f) == 0);
```

Comment:
>test6_e [](http://example.com/codeflow?start=29&length=7)

would be too awesome to return "test6_e OR test6_f". I suppose that is not doable, is it?

Reply:
It actually is doable, we'd have to do the whole loop through all the enum values but we could do it.
I suggest doing it as a separate PR

---
In reply to: [632130888](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632130888) [](ancestors = 632130888)